plugins/org.eclipse.n4js.generator.headless/src/org/eclipse/n4js/generator/headless/BuildSetComputer.java

```diff
@@ -146,9 +166,30 @@ public BuildSet createAllProjectsBuildSet(List<File> searchPaths) throws N4JSCom
 	 */
 	public BuildSet createBuildSet(List<File> searchPaths, List<File> projectPaths, List<File> singleSourceFiles)
 			throws N4JSCompileException {
+		printCompileArguments(searchPaths, projectPaths, singleSourceFiles);
 		return collectProjects(searchPaths, projectPaths, singleSourceFiles);
 	}
 
+	/**
+	 * Prints out some debug information about the user-provided compilation arguments (only if {@link #logger} is
+	 * configured in debug mode.
+	 *
+	 * @param searchPaths
+	 *            where to search for dependent projects.
+	 * @param projectPaths
+	 *            the projects to compile. the base folder of each project must be provided.
+	 * @param singleSourceFiles
+	 *            if non-empty limit compilation to the sources files listed here
+	 */
+	private void printCompileArguments(List<File> searchPaths, List<File> projectPaths, List<File> singleSourceFiles) {
```

Comment:
those are not compile arguments, those are derived configurations. 
Maybe name it `printBuildSetConfiguration` ?